connectors/sources/onedrive.py

```diff
@@ -87,6 +92,75 @@ class NotFound(Exception):
     pass
 
 
+class OneDriveAdvancedRulesValidator(AdvancedRulesValidator):
+    RULES_OBJECT_SCHEMA_DEFINITION = {
+        "type": "object",
+        "properties": {
+            "skipFilesWithExtensions": {
+                "type": "array",
+                "minItems": 1,
+                "items": {"type": "string"},
+            },
+            "parentPathPattern": {"type": "string", "minLength": 1},
+            "userMailAccounts": {
+                "type": "array",
+                "minItems": 1,
+                "items": {"type": "string", "format": "email", "minLength": 1},
+            },
+        },
+        "minProperties": 1,
+        "additionalProperties": False,
+    }
+
+    SCHEMA_DEFINITION = {"type": "array", "items": RULES_OBJECT_SCHEMA_DEFINITION}
+    SCHEMA = fastjsonschema.compile(definition=SCHEMA_DEFINITION)
+
+    def __init__(self, source):
+        self.source = source
+
+    async def validate(self, advanced_rules):
+        if len(advanced_rules) == 0:
+            return SyncRuleValidationResult.valid_result(
+                SyncRuleValidationResult.ADVANCED_RULES
+            )
+
+        return await self._remote_validation(advanced_rules)
+
+    @retryable(
+        retries=RETRIES,
+        interval=RETRY_INTERVAL,
+        strategy=RetryStrategy.EXPONENTIAL_BACKOFF,
+    )
+    async def _remote_validation(self, advanced_rules):
```

Comment:
Do we really need `_remote_validation`? Even though customers specify a non-existing user `foo`, we just don't index any document.